src/wp-includes/class-wp-user.php

```diff
@@ -288,6 +299,10 @@ public function __isset( $key ) {
 			$key = 'ID';
 		}
 
+		if ( in_array( $key, array( 'caps', 'allcaps', 'roles' ), true ) ) {
+			return isset( $this->$key );
```

Comment:
Shouldn't it always return `true` here since they will always be populated magically?

```suggestion
			return true;
```